New/task

Jeanstaquet opened the pull request:
<!-- This is a pull request template, you do not need to uncomment or remove the comments, they won't show up in the PR text. -->

<!-- Your Pull Request name should start with one of the following tags
  [NEW] For new features
  [IMPROVE] For an improvement (performance or little improvements) in existing features
  [FIX] For bug fixes that affect the end-user
  [BREAK] For pull requests including breaking changes
  Chore: For small tasks
  Doc: For documentation
-->

<!-- Checklist!!! If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. 
  - I have read the Contributing Guide - https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat doc
  - I have signed the CLA - https://cla-assistant.io/RocketChat/Rocket.Chat
  - Lint and unit tests pass locally with my changes
  - I have added tests that prove my fix is effective or that my feature works (if applicable)
  - I have added necessary documentation (if applicable)
  - Any dependent changes have been merged and published in downstream modules
-->

## Proposed changes (including videos or screenshots)
<!-- CHANGELOG -->
<!--
  Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
  If it fixes a bug or resolves a feature request, be sure to link to that issue below.
  This description will appear in the release notes if we accept the contribution.
-->

<!-- END CHANGELOG -->

## Issue(s)
<!-- Link the issues being closed by or related to this PR. For example, you can use #594 if this PR closes issue number 594 -->

## Steps to test or reproduce
<!-- Mention how you would reproduce the bug if not mentioned on the issue page already. Also mention which screens are going to have the changes if applicable -->

## Further comments
<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

Jeanstaquet committed:
Adding the new types for the taskRoom

Jeanstaquet committed:
Going to add the services

Jeanstaquet committed:
The task room can be created'

Jeanstaquet committed:
Removing the teammain

Jeanstaquet committed:
Adding the definition for the groups

Jeanstaquet committed:
Adding the correct group type

Jeanstaquet committed:
Adding the taskRoomId for the new rooms

Jeanstaquet committed:
Adding the comment to the routes

Jeanstaquet committed:
Adding the taksRoomId at various places

Jeanstaquet committed:
Correcting the bug of the room lists

Jeanstaquet committed:
Adding minimal action the taskRoom

Jeanstaquet committed:
Going to create the new task messages

Jeanstaquet committed:
Adding new files

Jeanstaquet committed:
New route: taskDetails

Jeanstaquet committed:
Adding the endpoint, going to add the modal to test the endpoint

Jeanstaquet committed:
bug new endpoint

Jeanstaquet committed:
Correcting the taskRoom.taskDetails bug

Jeanstaquet committed:
Adding a new modal for the task details

Jeanstaquet committed:
Adding a task MessageActions on the menu&message group

Jeanstaquet committed:
Adding the handlers for the modal

Jeanstaquet committed:
Correcting the bugs of the modal, going to implement fuselage for the taskModal

Jeanstaquet committed:
convert the html of the modal into fuselage

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
going to add a new endpoint

Jeanstaquet committed:
Adding the code for the update endpoint

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
Adding a new collection

Jeanstaquet committed:
Adding the new model

Jeanstaquet committed:
Adding the new template for the rooms

Jeanstaquet committed:
Adding the possibility to create a task raw

Jeanstaquet committed:
Adding new settings

Jeanstaquet committed:
Bug room manager

Jeanstaquet committed:
Adding the getHistory for the taskRoom

Jeanstaquet committed:
Adding the raw possibility to retrieves tasks and to add tasks

Jeanstaquet committed:
Going to implement thread header on tasks

Jeanstaquet committed:
Adding the replies to the tasks

Jeanstaquet committed:
Adding the threads messages

Jeanstaquet committed:
We can send messages with task headers

Jeanstaquet committed:
adding a new endpoint

Jeanstaquet committed:
Adding the modal & correcting db field for the tasks

Jeanstaquet committed:
Adding the reply button on the tasks

Jeanstaquet committed:
Adapting callback functions for threads msg

Jeanstaquet committed:
Adapting msg callbacks to the tasks

Jeanstaquet committed:
Adding the minimal afterSaveTask callbacks

Jeanstaquet committed:
Adding new verifications to create a task

Jeanstaquet committed:
Adding a Urls parser function for tasks

Jeanstaquet committed:
Adding small improvement on the UI

Jeanstaquet committed:
Adding a new modal to create tasks, moving some functions there

Jeanstaquet committed:
Adding the task deleting when the room is deleted

Jeanstaquet committed:
Addign comments

Jeanstaquet committed:
Addign comments

Jeanstaquet committed:
Going to add the Roommanager

Jeanstaquet committed:
Adding a new box fuselage for the room

Jeanstaquet committed:
Adding an overflow for tasks container

Jeanstaquet committed:
Adding the permissions

Jeanstaquet committed:
adding the scrollbar event

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
adding the front end db for the tasks

ostjen commented on app/api/server/v1/commands.js:
```diff
@@ -64,7 +64,7 @@ API.v1.addRoute('commands.run', { authRequired: true }, {
 			return API.v1.failure('The parameters for the command must be a single string.');
 		}
-
+		console.log('run');
```
remove

ostjen commented on app/api/server/v1/taskRoom.ts:
```diff
@@ -0,0 +1,293 @@
+API.v1.addRoute('taskRoom.create', { authRequired: true }, {
+	post() {
+		//  Permission_taskRoom
+		if (!hasPermission(this.userId, 'create-team')) {
```
create-taskRoom?

Jeanstaquet replied:
Indeed thanks

ostjen commented on app/lib/server/functions/parseUrlsInTask.js:
```diff
@@ -0,0 +1,22 @@
+import { Markdown } from '../../../markdown/server';
+
+export const parseUrlsInTask = (task) => {
```
is there a difference between this and parseUrlsInMessage.?

Jeanstaquet replied:
Indeed, it was unnecessarily duplicated code, I modified parseURLInMessage slightly to pass the tasks

ostjen commented on app/api/server/v1/chat.js:
```diff
@@ -86,6 +86,25 @@ API.v1.addRoute('chat.getMessage', { authRequired: true }, {
 			return API.v1.failure('The "msgId" query parameter must be provided.');
 		}
 
+		if (this.queryParams.msgId.includes('taskId:')) {
```
I think it's better to create a chat.getTask, or is there any drawback that I'm not seeing?

Jeanstaquet replied:
The problem I was trying to solve is that when we want to open a thread with a task as a thread header, we have to use the 
FlowRouter.setParams({
			tab: 'thread',
			context: `msgId`,
});
This action will call chat.getMessages to get the threadHeader message and the thread messages. I had to find a way with the FlowRouter to get the information if the threadHeader was a message or a task.

Jeanstaquet replied:
@ostjen Maybe you have a better solution, I think the solution I made is not optimal

Jeanstaquet replied:
Problem solved

ostjen commented on app/lib/server/methods/filterBadWords.js:
```diff
@@ -16,13 +16,13 @@ callbacks.add('beforeSaveMessage', function(message) {
 		}
 		const filter = new Filter(options);
-
+		console.log('ed1');
```
remove

Jeanstaquet committed:
-----------RoomManager modifications--------------

Jeanstaquet committed:
Adding getMoreTask on the RoomHistoryManager

Jeanstaquet committed:
Correcting the bug of getMore

Jeanstaquet committed:
adding roomManager features for the taksRoom

Jeanstaquet committed:
TO do : correcting the bug of the wrapper

Jeanstaquet committed:
Adding the taskStream of the record.streamActive

ostjen commented on app/lib/server/functions/sendTask.js:
```diff
@@ -0,0 +1,260 @@
+ * is going to be rendered in the href attribute of a
+ * link.
+ */
+const ValidFullURLParam = Match.Where((value) => {
```
if those validators are the same ones as in [sendMessage.js](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/lib/server/functions/sendMessage.js) it would be cool to export those into a separated file to avoid duplicated code.

Jeanstaquet replied:
Indeed, i adapted the valideteMessage from sendMessage and exported it in the sendTask file.

ostjen commented on app/lib/server/methods/getMessages.js:
```diff
@@ -8,7 +8,7 @@ Meteor.methods({
 
 		const cache = {};
-
+		console.log(messages);
```
remove

ostjen commented on app/spotify/lib/spotify.js:
```diff
@@ -17,7 +17,7 @@ export const createSpotifyBeforeSaveMessageHandler = () => (message) => {
 	let changed = false;
 
-	process(message, message.msg, (message, msgParts, index, part) => {
+	process(message, message.msg || message.title, (message, msgParts, index, part) => {
```
why modify spotify.js?

Jeanstaquet replied:
It because the beforeSaveMessage callback calls the createSpotifyBeforeSaveMessageHandler each time a message is sent. And the beforeSaveMessage callbacks is also borrowed by the task creation, so I had to adapt the callback.

Jeanstaquet replied:
It is via the beforeSaveMessage that the createSpotifyBeforeSaveMessageHandler is called each time a message is sent. And the beforeSaveMessage callbacks is also borrowed by the task creation, so I had to adapt the callback.

ostjen commented on app/ui-utils/client/lib/openRoom.js:
```diff
@@ -32,6 +32,7 @@ export const waitUntilRoomBeInserted = async (type, rid) => new Promise((resolve
 
 
 NewRoomManager.on('changed', (rid) => {
+	console.log('newRoom');
```
remove

ostjen commented on app/ui/client/lib/chatMessages.js:
```diff
@@ -247,7 +247,7 @@ export class ChatMessages {
 	async send(event, { rid, tmid, value, tshow }, done = () => {}) {
 		const threadsEnabled = settings.get('Threads_enabled');
-
+		console.log(event);
```
remove

ostjen commented on app/ui/client/views/app/room.js:
```diff
@@ -189,6 +189,7 @@ Template.roomOld.helpers({
 		return state.get('subscribed');
 	},
 	messagesHistory() {
+		console.log('msghist');
```
remove

ostjen commented on server/features/EmailInbox/EmailInbox_Outgoing.ts:
```diff
@@ -140,7 +140,7 @@ callbacks.add('beforeSaveMessage', function(message: any, room: any) {
 		return message;
 	}
-
+	console.log('run');
```
remove

ostjen commented on app/threads/client/lib/normalizeThreadTitle.js:
```diff
@@ -26,6 +27,9 @@ export const normalizeThreadTitle = ({ ...message }) => {
 			roomTemplate: ({ prefix, mention }) => `${ prefix }<strong> ${ mention } </strong>`,
 		});
 		const { html } = emojiParser({ html: filteredMessage });
+		if (message.title) {
```
I guess it would be cleaner to do something like
```
const parsedInstace = message.title ? instance.parse() : instance.parse()
return parsedInstance
```

Jeanstaquet replied:
Corrected, thanks

ostjen commented on client/views/room/providers/RoomProvider.tsx:
```diff
@@ -28,6 +29,7 @@ const RoomProvider = ({ rid, children }: Props): JSX.Element => {
 
 	useEffect(() => {
 		RoomManager.open(rid);
+		console.log('openRM');
```
remove

ostjen commented on app/ui-message/client/messageThread.js:
```diff
@@ -22,6 +22,7 @@ const findParentMessage = (() => {
 	const get = async (tmid) => {
 		getMessages();
 		const messages = await pending;
+		console.log(messages);
```
remove

ostjen commented on app/ui-message/client/messageThread.js:
```diff
@@ -11,7 +11,7 @@ const findParentMessage = (() => {
 	let resolve;
 	let pending = new Promise((r) => { resolve = r; });
-
+	console.log('1222222');
```
remove

ostjen commented on app/ui-message/client/message.js:
```diff
@@ -58,6 +58,7 @@ Template.message.helpers({
 	},
 	unread() {
 		const { msg, subscription } = this;
+		console.log(subscription?.tunread?.includes(msg._id), subscription);
```
remove

ostjen commented on app/lib/server/methods/filterBadWords.js:
```diff
@@ -16,13 +16,13 @@ callbacks.add('beforeSaveMessage', function(message) {
 		}
 
-		message.msg = filter.clean(message.msg);
+		message.msg ? message.msg = filter.clean(message.msg) : message.title = filter.clean(message.title);
```
I think it'd be easier to read something like
```
message.msg = message.msg ? filter.clean(message.msg) : filter.clean(message.title)
```

Jeanstaquet replied:
In this statement I wanted to make sure that if msg exists, we assign msg = filter.clean(message.msg). Whereas if it did not exist, we assign title = filter.clean(message.title)

ostjen replied:
you're right, i misread it

ostjen commented on app/lib/server/methods/sendMessage.js:
```diff
@@ -63,8 +63,8 @@ export function executeSendMessage(uid, message) {
 		const parentMessage = Messages.findOneById(message.tmid);
-		message.tmid = parentMessage.tmid || message.tmid;
-		rid = parentMessage.rid;
+		message.tmid = (parentMessage && parentMessage.tmid) || message.tmid;
```
why?

Jeanstaquet replied:
The problem is that tasks are not messages under the hood. However, tasks can become header threads. And so the parentMessage became undefined. 

Maybe i should find a way to find a parentTask as well depending on whether we add a message under a task.

ostjen commented on app/lib/server/methods/sendMessage.js:
```diff
@@ -63,8 +63,8 @@ export function executeSendMessage(uid, message) {
-		message.tmid = parentMessage.tmid || message.tmid;
-		rid = parentMessage.rid;
+		message.tmid = (parentMessage && parentMessage.tmid) || message.tmid;
+		rid = (parentMessage && parentMessage.rid) || message.rid;
```
same as above

ostjen commented on client/views/taskRoom/components/TaskRoom/TaskRoom.js:
```diff
@@ -0,0 +1,120 @@
+	};
+
+	const tasksWrapper = useRef(null);
+	console.log(tasksWrapper);
```
remove

ostjen commented on client/views/taskRoom/components/TaskRoom/TaskRoom.js:
```diff
@@ -0,0 +1,120 @@
+	useEffect(() => {
+		const fetchData = async () => {
+			const d = await ChatTask.find({ rid }).fetch();
+			console.log(d);
```
remove

ostjen commented on app/mentions/server/Mentions.js:
```diff
@@ -48,8 +48,13 @@ export default class MentionsServer extends MentionsParser {
-	getUsersByMentions({ msg, rid, u: sender }) {
-		let mentions = this.getUserMentions(msg);
+	getUsersByMentions({ msg, title, rid, u: sender }) {
+		let mentions;
+		if (msg) {
+			mentions = this.getUserMentions(msg);
+		} else {
+			mentions = this.getUserMentions(title);
```
```
let mentions = msg ? this.getUserMentions(msg) : this.getUserMentions(title);
```
?

Jeanstaquet replied:
Indeed

ostjen commented on app/ui-utils/client/lib/RoomHistoryManager.js:
```diff
@@ -191,7 +258,7 @@ export const RoomHistoryManager = new class extends Emitter {
 		const showMessageInMainThread = getUserPreference(Meteor.userId(), 'showMessageInMainThread', false);
 		const result = await call('loadHistory', rid, ts, limit, ls, showMessageInMainThread);
-
+		console.log('loadHistory ->');
```
remove

ostjen commented on app/ui-utils/client/lib/RoomHistoryManager.js:
```diff
@@ -181,6 +247,7 @@ export const RoomHistoryManager = new class extends Emitter {
 		let typeName = undefined;
 
 		const subscription = ChatSubscription.findOne({ rid });
+		console.log(subscription);
```
remove

ostjen commented on app/ui-utils/client/lib/RoomHistoryManager.js:
```diff
@@ -82,13 +98,48 @@ export const upsertMessage = async ({ msg, subscription, uid = Tracker.nonreacti
 export function upsertMessageBulk({ msgs, subscription }, collection = ChatMessage) {
 	const uid = Tracker.nonreactive(() => Meteor.userId());
 	const { queries } = ChatMessage;
+	console.log('upsertMessageBulk');
```
remove

ostjen commented on app/ui-utils/client/lib/RoomHistoryManager.js:
```diff
@@ -98,6 +149,19 @@ export function upsertMessageBulk({ msgs, subscription }, collection = ChatMessa
+export function upsertTaskBulk({ tasks, subscription }, collection = ChatTask) {
+	const uid = Tracker.nonreactive(() => Meteor.userId());
+	const { queries } = ChatTask;
+	console.log('upsertMessageBulk');
```
remove

ostjen commented on app/ui-utils/client/lib/RoomHistoryManager.js:
```diff
@@ -141,6 +206,7 @@ export const RoomHistoryManager = new class extends Emitter {
 
 	run(fn) {
 		const difference = differenceInMilliseconds(new Date(), this.lastRequest);
+		console.log('grun');
```
remove

ostjen commented on app/ui-utils/client/lib/RoomHistoryManager.js:
```diff
@@ -120,13 +184,14 @@ export const RoomHistoryManager = new class extends Emitter {
 	async queue() {
 		return new Promise((resolve) => {
 			const requestId = uuidv4();
+			console.log('queue');
```
remove

ostjen commented on app/ui-utils/client/lib/RoomHistoryManager.js:
```diff
@@ -120,13 +184,14 @@ export const RoomHistoryManager = new class extends Emitter {
 			};
 		}
-
+		console.log('getRoom, RoomHistoryManager');
```
remove

ostjen commented on app/ui-utils/client/lib/RoomHistoryManager.js:
```diff
@@ -50,6 +50,22 @@ export const waitUntilWrapperExists = async (selector = '.messages-box .wrapper'
+		childList: true,
+		subtree: true,
+	});
+	console.log('waitUntilWrapperExists');
```
remove

ostjen commented on app/ui-utils/client/lib/RoomHistoryManager.js:
```diff
@@ -50,6 +50,22 @@ export const waitUntilWrapperExists = async (selector = '.messages-box .wrapper'
 		childList: true,
 		subtree: true,
 	});
+	console.log('waitUntilWrapperExists');
```
remove

ostjen commented on client/components/Message/Actions/Actions.tsx:
```diff
@@ -22,9 +22,9 @@ const Actions: FC<{ actions: Array<ActionOptions>; runAction: RunAction; mid: st
-			{actions.map((action) => (
-				<Action runAction={runAction} key={action.id} {...action} />
-			))}
+			{actions !== undefined &&
+				actions[0] &&
+				actions.map((action) => <Action runAction={runAction} key={action.id} {...action} />)}
```
why?

Jeanstaquet committed:
WIP reactive room with the RoomHistoryManager

Jeanstaquet committed:
-------------Going to add the task stream-----------------

Jeanstaquet committed:
adding the changes for the tasks in the notifications modules

Jeanstaquet committed:
Adding the settings for the watch feature

Jeanstaquet committed:
Correcting services TaskRoom

Jeanstaquet committed:
correction of a bug on the taskroom watch feature

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
Correcting the create new taskRoom bug & adding some cleaning for the modal

Jeanstaquet committed:
going to solve the promise error for unsubscribeFromRoomMessages

Jeanstaquet committed:
Adding the two stream for the taks room

Jeanstaquet committed:
going to merge two differents streams

Jeanstaquet committed:
.

Jeanstaquet committed:
cleaning + redifining msg & task streams

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
Removing console logs

Jeanstaquet committed:
More cleaning

Jeanstaquet committed:
Adding some cleaning, modification of the validateTask

Jeanstaquet committed:
Removing useless code & export validateMessage into sendTask

Jeanstaquet committed:
Adding the delete endpoint & feature

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
cleaning

Jeanstaquet committed:
test react meteor-data

Jeanstaquet committed:
adding a new hook from meteor

Jeanstaquet committed:
Correct a bug related with the threadlist

Jeanstaquet committed:
Adding the follow feature

Jeanstaquet committed:
cleaning

Jeanstaquet committed:
Adding the i18n keys

Jeanstaquet committed:
Adding a fix for the sorting of the tasks

Jeanstaquet committed:
cleaning

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
Adding a task & correcting the loadhistory bug

Jeanstaquet committed:
cleaning

Jeanstaquet committed:
adding test

Jeanstaquet committed:
adding translations in the i18n file

Jeanstaquet committed:
Correcting a bug

Jeanstaquet committed:
removing unused tabBar elements

Jeanstaquet committed:
Improving the getMessage endpoint for task header

Jeanstaquet commented on app/lib/server/functions/sendTask.js:
```diff
@@ -0,0 +1,110 @@
+
+		/*
+		Defer other updates as their return is not interesting to the user
+		*/
```
@ostjen Here should be the code for the app engine, which is not yet made for tasks. Do I leave the code as a comment or do I delete the code as a comment?

Jeanstaquet commented on app/lib/server/functions/sendTask.js:
```diff
@@ -0,0 +1,110 @@
+		task.unread = true;
+	}
+
+	// For the Rocket.Chat Apps :)
+	// if (Apps && Apps.isLoaded()) {
+	// 	const prevent = Promise.await(Apps.getBridges().getListenerBridge().messageEvent('IPreMessageSentPrevent', message));
+	// 	if (prevent) {
+	// 		if (settings.get('Apps_Framework_Development_Mode')) {
+	// 			console.log('A Rocket.Chat App prevented the message sending.', message);
+	// 		}
+
+	// 		return;
+	// 	}
+
+	// 	let result;
+	// 	result = Promise.await(Apps.getBridges().getListenerBridge().messageEvent('IPreMessageSentExtend', message));
+	// 	result = Promise.await(Apps.getBridges().getListenerBridge().messageEvent('IPreMessageSentModify', result));
+
+	// 	if (typeof result === 'object') {
+	// 		message = Object.assign(message, result);
+
+	// 		// Some app may have inserted malicious/invalid values in the message, let's check it again
+	// 		validateMessage(message, room, user);
+	// 	}
```
@ostjen Here should be the code for the app engine, which is not yet made for tasks. What do you think do I leave the code as a comment or do I delete the code as a comment? I think it can be a good idea to leave it as it is.

Jeanstaquet committed:
adding a function for the threads updating for tasks header

Jeanstaquet committed:
Correcting the display of task title on the threadlist

Jeanstaquet committed:
improving permissions for deleteTask and taskhistory endpoints

Jeanstaquet committed:
correcting global permissions for tasks

Jeanstaquet committed:
correcting sendTask method

Jeanstaquet committed:
Adding the meteor method for createTask

Jeanstaquet committed:
Correcting the follow button feature on thread with task eader

Jeanstaquet committed:
Improving the thread messages with a task header

Jeanstaquet committed:
bug in the thread component

Jeanstaquet committed:
Correcting bug for the threadlist in a taskRoom

Jeanstaquet committed:
cleaning

Jeanstaquet committed:
Cleaning the useStreamUpdateForMEssagesList hook

Jeanstaquet committed:
Adding a possible taskRoomHandler

Jeanstaquet commented on app/ui/client/lib/taskRoomUtils.js:
```diff
@@ -0,0 +1,78 @@
+// Back-end
+const sendTask = () => {
+
+};
+
+const processUpdateTask = () => {
+
+};
+
+const updateTask = () => {
+
+};
+
+const processDeleteTask = () => {
+
+};
+
+const deleteTask = () => {
+
+};
+
+export const NewTaskRoom = {
+	async createTask(rid, cb) {
+		// Permissions
+		processCreateTask();
+		sendTask();
+	},
+
+	updateTask(cb) {
+		// Permissions
+		processUpdateTask();
+		updateTask();
+	},
+
+	deleteTask(cb) {
+		// Permissions
+		processDeleteTask();
+		deleteTask();
+	},
+};
```
@ostjen I was thinking of importing this into the taskRoom to manage the methdos of the tasks. From here, we call the MiniMongo collection from the front-end, and the endpoint from the back-end

Jeanstaquet committed:
Bug in the newCreateTask

Jeanstaquet committed:
go back

Jeanstaquet committed:
Revert "go back"

This reverts commit 39dd41bc26c467304b6991f2576454f6deadf4bd.

Jeanstaquet committed:
Revert "Bug in the newCreateTask"

This reverts commit a8b1e19d3c4920cb38f9680bded7087653efa8eb.

Jeanstaquet committed:
Revert "Adding a possible taskRoomHandler"

This reverts commit cc1952b9a08bf8085c948a281eac738c34e6787f.

Jeanstaquet committed:
cleaning

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
Correcting tests

Jeanstaquet committed:
Cleaning test code for the taskRoom

Jeanstaquet committed:
cleaning

Jeanstaquet committed:
Adding tests

Jeanstaquet committed:
Updating permissions and adding tests taskUpdate

Jeanstaquet committed:
Correcting permissions endpoint taskRoom

Jeanstaquet committed:
adding a bunch of tests, correcting user permissions

Jeanstaquet committed:
adding tests for .details .delete

Jeanstaquet committed:
Improving the thread component inside a taskRoom

Jeanstaquet committed:
Adding a new methods for following the task inside the thread component

Jeanstaquet committed:
Correcting create taskRoomModal

Jeanstaquet committed:
clean the package-lock

Jeanstaquet committed:
cleaning extra spaces

Jeanstaquet committed:
Correcting the read threads bug

Jeanstaquet committed:
Correcting the bug of notifications

Jeanstaquet committed:
cleaning

Jeanstaquet committed:
resolving conflicts

Jeanstaquet committed:
Correcting ActiveUsersSections.js

Jeanstaquet committed:
Merge branch 'develop' into new/task

Jeanstaquet committed:
Update updateTask.js

Jeanstaquet committed:
Update CreateRoomList.js

Jeanstaquet committed:
Update taskRoom.js

Jeanstaquet committed:
Update withData.js

Jeanstaquet committed:
resolving conflicts

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
Merge branch 'develop' into new/task

Jeanstaquet committed:
Cleaning

Jeanstaquet committed:
Merge branch 'new/task' of https://github.com/Jeanstaquet/Rocket.Chat into new/task

Jeanstaquet committed:
Cleaning types

Jeanstaquet committed:
Corrections

Jeanstaquet committed:
Correcting the parseulr for new messages

Jeanstaquet committed:
Merge branch 'develop' into new/task

Jeanstaquet committed:
Merge branch 'develop' into new/task

Jeanstaquet committed:
correcting beforesavecallback for tasks

Jeanstaquet committed:
Merge branch 'new/task' of https://github.com/Jeanstaquet/Rocket.Chat into new/task

Jeanstaquet committed:
Improving

Jeanstaquet committed:
Improving getThreadsMessages

Jeanstaquet committed:
improving getThreadMessages

Jeanstaquet committed:
Clean

Jeanstaquet committed:
Improving getThreadMessages for tasks

Jeanstaquet committed:
cleaning package

Jeanstaquet committed:
improving file data streams for the taskRoom

Jeanstaquet committed:
eslint

sampaiodiego committed:
Bump version to 3.18.0-rc.0

sampaiodiego committed:
Merge remote-tracking branch 'origin/develop' into release-candidate

sampaiodiego committed:
Bump version to 3.18.0-rc.1

sampaiodiego committed:
Merge remote-tracking branch 'origin/develop' into release-candidate

sampaiodiego committed:
Bump version to 3.18.0-rc.2

sampaiodiego committed:
Merge remote-tracking branch 'origin/develop' into release-candidate

sampaiodiego committed:
Bump version to 3.18.0-rc.3

sampaiodiego committed:
Merge remote-tracking branch 'origin/develop' into release-candidate

sampaiodiego committed:
Bump version to 3.18.0-rc.4

sampaiodiego committed:
Bump version to 3.18.0

sampaiodiego committed:
Merge pull request #23066 from RocketChat/release-3.18.0

Release 3.18.0

Jeanstaquet committed:
Preparing for merge conflicts

Jeanstaquet committed:
Resolving conflicts

Jeanstaquet committed:
Merge branch 'develop' into new/task

sampaiodiego committed:
Chore: Remove wrong usages of `Meteor.wrapAsync` (#23079)

Jeanstaquet committed:
Merge branch 'develop' into new/task

sampaiodiego committed:
Merge remote-tracking branch 'origin/master' into develop-sync

sampaiodiego committed:
Bump version to 4.0.0

sampaiodiego committed:
Merge pull request #23086 from RocketChat/develop-sync

Merge master into develop & Set version to 4.0.0

lingohub[bot] committed:
Language update from LingoHub 🤖 (#23061)

Project Name: Rocket.Chat
Project Link: https://translate.lingohub.com/rocketchat/dashboard/rocket-dot-chat
User: Robot LingoHub

Easy language translations with LingoHub 🚀

Co-authored-by: Robot LingoHub <robot@lingohub.com>

dependabot[bot] committed:
Bump actions/stale from 3.0.19 to 4 (#22673)

Bumps [actions/stale](https://github.com/actions/stale) from 3.0.19 to 4.
- [Release notes](https://github.com/actions/stale/releases)
- [Changelog](https://github.com/actions/stale/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/stale/compare/v3.0.19...v4)

---
updated-dependencies:
- dependency-name: actions/stale
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @types/express from 4.17.12 to 4.17.13 in /ee/server/services (#22598)

Bumps [@types/express](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express) from 4.17.12 to 4.17.13.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/express)

---
updated-dependencies:
- dependency-name: "@types/express"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @types/cookie from 0.4.0 to 0.4.1 in /ee/server/services (#22600)

Bumps [@types/cookie](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/cookie) from 0.4.0 to 0.4.1.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/cookie)

---
updated-dependencies:
- dependency-name: "@types/cookie"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

graywolf336 committed:
Chore: Environmental variable for marketplace url (#22922)

dependabot[bot] committed:
Bump object-path from 0.11.5 to 0.11.6 (#23088)

Bumps [object-path](https://github.com/mariocasciaro/object-path) from 0.11.5 to 0.11.6.
- [Release notes](https://github.com/mariocasciaro/object-path/releases)
- [Commits](https://github.com/mariocasciaro/object-path/commits)

---
updated-dependencies:
- dependency-name: object-path
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Jeanstaquet committed:
Merge branch 'develop' into new/task

Jeanstaquet committed:
Update taskDetailsModal.js

Jeanstaquet commented:
a
